back/engines/commercial/bulk_import_ideas/app/services/bulk_import_ideas/import_ideas_service.rb

```diff
@@ -0,0 +1,216 @@
+# frozen_string_literal: true
+
+module BulkImportIdeas
+  class Error < StandardError
+    def initialize(key, params = {})
+      super()
+      @key = key
+      @params = params
+    end
+
+    attr_reader :key, :params
+  end
+
+  class ImportIdeasService
+    DEFAULT_MAX_IDEAS = 500
+
+    def import_ideas(idea_rows, max_ideas: DEFAULT_MAX_IDEAS)
+      raise Error.new 'bulk_import_ideas_maximum_ideas_exceeded', value: max_ideas if idea_rows.size > max_ideas
+
+      ActiveRecord::Base.transaction do
+        idea_rows.each do |idea_row|
+          idea = import_idea idea_row
+          Rails.logger.info { "Created #{idea.id}" }
+        end
+      end
+    end
+
+    def generate_example_xlsx
+      XlsxService.new.hash_array_to_xlsx [
+        {
+          'Title_nl-BE' => 'Mijn idee titel',
+          'Title_fr-BE' => 'Mon idée titre',
+          'Body_nl-BE' => 'Mijn idee inhoud',
+          'Body_fr-BE' => 'Mon idée contenu',
+          'Email' => 'moderator@citizenlab.co',
+          'Project' => 'Project 1',
+          'Phase' => 1,
+          'Image URL' => 'https://res.cloudinary.com/citizenlabco/image/upload/v1548847594/image_v8imrf.png',
+          'Date (dd-mm-yyyy)' => '2022-07-18',
+          'Latitude' => 50.5035,
+          'Longitude' => 6.0944,
+          'Location Description' => 'Panorama sur les Hautes Fagnes / Hohes Venn'
+        }
+      ]
+    end
+
+    def xlsx_to_idea_rows(xlsx)
+      xlsx.map do |xlsx_row|
+        idea_row = {}
+
+        title_multiloc = {}
+        body_multiloc  = {}
+        xlsx_row.each do |key, value|
+          next unless key.include? '_'
+
+          field, locale = key.split '_'
+          case field
+          when 'Title'
+            title_multiloc[locale] = value
+          when 'Body'
+            body_multiloc[locale] = value
+          end
+        end
+
+        idea_row[:title_multiloc]       = title_multiloc
+        idea_row[:body_multiloc]        = body_multiloc
+        idea_row[:topic_titles]         = (xlsx_row['Topics'] || '').split(';').map(&:strip).select { |topic| topic }
```

Comment:
I guess this is what you mean?
```suggestion
        idea_row[:topic_titles]         = (xlsx_row['Topics'] || '').split(';').select(&present?)
```